frontend/src/scenes/plugins/tabs/installed/InstalledTab.tsx

```diff
@@ -37,7 +84,48 @@ export function InstalledTab(): JSX.Element {
                     ? 'Check again'
                     : 'Check for updates'}
             </Button>
-        ) : null
+        ) : (
+            <></>
+        )
+
+    const canRearrange = user?.plugin_access.configure && enabledPlugins.length > 1
+
+    const rearrangingButtons = rearranging ? (
+        <Space>
+            <Button type="primary" loading={loading} onClick={() => savePluginOrders(temporaryOrder)}>
+                Save Order
+            </Button>
+            <Button type="default" onClick={cancelRearranging}>
+                Cancel
+            </Button>
+        </Space>
+    ) : (
+        <></>
+    )
+
+    const onSortEnd = ({ oldIndex, newIndex }: { oldIndex: number; newIndex: number }): void => {
+        if (oldIndex === newIndex) {
+            return
+        }
+
+        const move = (arr: PluginTypeWithConfig[], from: number, to: number): { id: number; order: number }[] => {
+            const clone = [...arr]
+            Array.prototype.splice.call(clone, to, 0, Array.prototype.splice.call(clone, from, 1)[0])
```

Comment:
Why on `Array.prototype.splice` instead of just `clone`?

Reply:
Hmm, not sure I fully understand the question here :). 

We need to clone, since `.splice` alters the array. The `splice` code itself is from a react-sortable-hoc demo and it's the shortest accurate reordering code I've found some far.